esm/interface/custom-element-registry.js

```diff
@@ -146,11 +148,11 @@ export class CustomElementRegistry {
         for (const [key] of values)
           delete element[key];
 
-        setPrototypeOf(element, new Class(this.ownerDocument, ce));
+        setPrototypeOf(element, Class.prototype);
         customElements.set(element, {connected: isConnected});
 
-        for (const [key, value] of values)
-          element[key] = value;
```

Comment:
OK, I think I am getting what's going on ... so, if you move this part into the HTMLElement constructor it has the side effect that `onconnectedCallback` will be callbed **before** accessors / properties are set, which is a violation of the standard or not how *upgrade* works.

I'll try to think about the issue ... if I understood what it is: basically the `new Class` as prototype interfere with lit because it can't trap the element right away, only an instance that is *not* actually the current element ... right? Sorry if I am being slow